zerver/decorator.py

```diff
@@ -131,7 +131,7 @@ def wrapper(request, user_profile, *args, **kwargs):
 from zerver.lib.user_agent import parse_user_agent
 
 def get_client_name(request, is_json_view):
-    # type: (HttpRequest, bool) -> Text
+    # type: (HttpRequest, bool) -> Union[Text, List[Text]]
```

Comment:
There is an underlying assumption in a lot of Zulip's code (from what I have read) that eg `request.GET['client']` can only return `Text`. However, GET and POST are QueryDicts, which [return `[]` if the dict is empty](https://github.com/django/django/blob/master/django/utils/datastructures.py#L71).